base/src/ArchiveSpaceManager.cpp

```diff
@@ -0,0 +1,180 @@
+        {
+            if ((boost::filesystem::is_regular_file(entry)))
+            {
+                if (!(countFreq % mProps.samplingFreq))
+                {
+                    sample = (rand() % mProps.samplingFreq) + 1;
+                    inCount = 0;
+                }
+                if (inCount == sample)
+                {
+                    tempSize = boost::filesystem::file_size(entry);
+                    dirSize += tempSize * mProps.samplingFreq;
+                }
+                inCount++;
+                countFreq++;
+            }
+        }
+        if (inCount < mProps.samplingFreq)
+        {
+            dirSize = dirSize + (tempSize * inCount);
+        }
+        return dirSize;
+    }
+
+    boost::filesystem::path getOldestDirectory(boost::filesystem::path _cam)
+    {
+        for (const auto& camFolder : boost::filesystem::directory_iterator(_cam))
+        {
+            for (const auto& folder : boost::filesystem::recursive_directory_iterator(camFolder))
+            {
+                if (boost::filesystem::is_regular_file(folder))
+                {
+                    boost::filesystem::path p = folder.path().parent_path();
+                    return p;
+                }
+            }
+        }
+    };
+
+    void manageDirectory()
+    {
+        auto comparator = [](std::pair<boost::filesystem::path, uint32_t>& a, std::pair<boost::filesystem::path, uint32_t>& b) {return a.second < b.second; };
+        while (archiveSize > mProps.lowerWaterMark)
+        {
+            for (const auto& camFolder : boost::filesystem::directory_iterator(mProps.pathToWatch))
+            {
+                boost::filesystem::path oldHrDir = getOldestDirectory(camFolder);
+                foldVector.push_back({oldHrDir,boost::filesystem::last_write_time(oldHrDir) });
+            }
+            sort(foldVector.begin(), foldVector.end(), comparator); //Sorting the vector
+
+            uint32_t tempSize = 0;
+            boost::filesystem::path delDir = foldVector[0].first;
+            BOOST_LOG_TRIVIAL(info) <<"Deleting file : "<< delDir.string();
+            tempSize = estimateDirectorySize(delDir);
+            archiveSize = archiveSize - tempSize;
+            boost::filesystem::remove_all(delDir);
+            foldVector.clear();
+        }
+    }
+
+    uint32_t diskOperation()
+    {
+        archiveSize = estimateDirectorySize(mProps.pathToWatch);
+        if (archiveSize > mProps.upperWaterMark)
+        {
+            manageDirectory();
+        }
+        uint32_t tempSize = archiveSize;
+        archiveSize = 0;
+        return tempSize;
+    }
+    ArchiveSpaceManagerProps mProps;
+    uint32_t archiveSize = 0;
+    std::vector<std::pair<boost::filesystem::path, uint32_t>> foldVector;
+};
+
+
+ArchiveSpaceManager::ArchiveSpaceManager(ArchiveSpaceManagerProps _props)
+    :Module(TRANSFORM, "DiskspaceManager", _props)
```

Comment:
Kind shoud be SOURCE
The space manager is a free running thread which decides to exampine disk space periodically

Reply:
ArchiveSpaceManager is changed to source and test cases are refactored.